network/p2p/handler.go

```diff
@@ -58,6 +64,28 @@ func (NoOpHandler) CrossChainAppRequest(context.Context, ids.ID, time.Time, []by
 	return nil, nil
 }
 
+// ValidatorHandler drops messages from non-validators
+type ValidatorHandler struct {
+	Handler
+	ValidatorSet ValidatorSet
```

Comment:
I know we don't do this very well in the rest of the repo... but should we make this a minimal interface? Doesn't seem like we need any of the sampling functionality here.